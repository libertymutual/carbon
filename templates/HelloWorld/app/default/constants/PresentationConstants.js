// Presentation Constants

// Common Constants
export const CONTINUE = 'Continue';
export const CLOSE = 'Close';
export const OK = 'Ok';
export const YES = 'Yes';
export const NO = 'No';
export const GO_BACK = 'Go Back';

// Page Titles
export const PAGE_TITLE_HOME_SCREEN = 'Home';
export const PAGE_TITLE_CONTACT_SCREEN = 'Contact Us';
export const PAGE_TITLE_PROFILE_SCREEN = 'Profile';
export const PAGE_TITLE_ABOUT_SCREEN = 'About';
export const PAGE_TITLE_BIOMETRIC_SCREEN = 'Biometric Features';

// Home Screen Presentation Constants
export const HOMESCREEN_WELCOME = 'Welcome To The Default Feature Home Screen';
export const HOMESCREEN_INTRO = 'This is a sample home/landing page for a React Native Feature. This page uses conditional logic & a redux store to display the profile page button only when the user is logged in. In addition, the text and behaviour of the login button will change dependent on whether the user is currently logged in or logged out. It also allows you to test navigating to another React Native Feature separate from the Default Feature';
export const HOMESCREEN_BUTTON_LABEL_LOGIN = 'Login';
export const HOMESCREEN_BUTTON_LABEL_LOGOUT = 'Logout';
export const HOMESCREEN_BUTTON_LABEL_PROFILE = 'Profile';
export const HOMESCREEN_BUTTON_LABEL_OTHER_FEATURE = 'Another Feature';
export const HOMESCREEN_BUTTON_LABEL_WEB_FEATURE = 'Web Feature';
export const HOMESCREEN_BUTTON_LABEL_CONTACT_US = 'Contact Us';
export const HOMESCREEN_BUTTON_LABEL_ABOUT = 'About';
export const HOMESCREEN_BUTTON_LABEL_BIOMETRIC = 'Biometric Features Test Screen';
export const HOMESCREEN_BUTTON_LABEL_BUILD_TYPE = 'Get Build Type'

// Contact Screen Presentation Constants
export const CONTACT_WELCOME = 'Welcome To The Default Feature Contact Us Screen';

// Profile Screen Presentation Constants
export const PROFILE_WELCOME = 'Welcome To The Default Feature Profile Screen';
export const PROFILE_INTRO = 'This sample page allows you to test updating a persistent store, the value of which will be retained between app restarts';
export const PROFILE_COUNTER_VALUE = 'Counter Value:';
export const PROFILE_INCREMENT_COUNTER = 'Increment Counter';
export const PROFILE_DECREMENT_COUNTER = 'Decrement Counter';

// About Screen Presentation Constants
export const ABOUT_WELCOME = 'About This App';
export const ABOUT_INTRO = 'This is the About This App page';

// Biometric Screen Presentation Constants
export const BIOMETRIC_WELCOME = 'Biometric Features Test Screen';
export const BIOMETRIC_INTRO = 'Use this screen to test biometric features such as Fingerprint / Facial authorization';
export const BIOMETRIC_USERNAME = 'Username';
export const BIOMETRIC_PASSWORD = 'Passsword';
export const BIOMETRIC_BUTTON_LABEL_SAVE_CREDENTIALS = 'Save Credentials'
export const BIOMETRIC_BUTTON_LABEL_GET_CREDENTIALS = 'Get Credentials'
export const BIOMETRIC_BUTTON_LABEL_HAS_CREDENTIALS = 'Check If Credentials Stored'
export const BIOMETRIC_BUTTON_LABEL_DELETE_CREDENTIALS = 'Delete Credentials'
export const BIOMETRIC_BUTTON_LABEL_CHECK_SUPPORTED = 'Check Device Support'
export const BIOMETRIC_BUTTON_LABEL_GET_BIOMETRIC_TYPE = 'Check Biometric Authentication Type\n(Face/Fingerprint)'
