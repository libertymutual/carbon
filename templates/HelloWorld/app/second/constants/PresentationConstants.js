// Presentation Constants

export const CONTINUE = 'Continue';
export const CLOSE = 'Close';
export const OK = 'Ok';
export const YES = 'Yes';
export const NO = 'No';
export const GO_BACK = 'Go Back';

// Page Titles
export const PAGE_TITLE_HOME_SCREEN = 'Second Feature Home';
export const PAGE_TITLE_ABOUT_SCREEN = 'About';

// Home Screen Presentation Constants
export const HOMESCREEN_WELCOME = 'Welcome To The Second Feature Home Screen';
export const HOMESCREEN_INTRO = 'This is an example of navigating to a completely separate React Native Feature\n\nIt also shows an example of sharing state between two separate React Native Features. The Counter Value below is the same counter value displayed/incremented/decremented on the Default Feature';
export const HOMESCREEN_COUNTER_VALUE = 'Counter Value:';
export const HOMESCREEN_BUTTON_LABEL_BACK_TO_DEFAULT_FEATURE = 'Back To Default Feature';
export const HOMESCREEN_BUTTON_LABEL_ABOUT = 'About';

// About Screen Presentation Constants
export const ABOUT_WELCOME = 'Welcome To The Second Feature About Screen';
export const ABOUT_BUTTON_LABEL_TOGGLE = 'Toggle Hello World Message';
export const ABOUT_HELLO_WORLD = 'Hello World';
export const ABOUT_GOODBYE_WORLD = 'Goodbye World';
