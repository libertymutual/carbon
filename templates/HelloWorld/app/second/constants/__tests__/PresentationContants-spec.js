import 'react-native';
import React from 'react';
import * as PresentationConstants from '../PresentationConstants';

describe('PresentationConstants tests', () => {

  test('CONTINUE returns correct value', () => {
    expect(PresentationConstants.CONTINUE).toMatchSnapshot();
  });

  test('CLOSE returns correct value', () => {
    expect(PresentationConstants.CLOSE).toMatchSnapshot();
  });

  test('OK returns correct value', () => {
    expect(PresentationConstants.OK).toMatchSnapshot();
  });

  test('YES returns correct value', () => {
    expect(PresentationConstants.YES).toMatchSnapshot();
  });

  test('NO returns correct value', () => {
    expect(PresentationConstants.NO).toMatchSnapshot();
  });

  test('GO_BACK returns correct value', () => {
    expect(PresentationConstants.GO_BACK).toMatchSnapshot();
  });

  test('PAGE_TITLE_HOME_SCREEN returns correct value', () => {
    expect(PresentationConstants.PAGE_TITLE_HOME_SCREEN).toMatchSnapshot();
  });

  test('PAGE_TITLE_ABOUT_SCREEN returns correct value', () => {
    expect(PresentationConstants.PAGE_TITLE_ABOUT_SCREEN).toMatchSnapshot();
  });

  test('HOMESCREEN_WELCOME returns correct value', () => {
    expect(PresentationConstants.HOMESCREEN_WELCOME).toMatchSnapshot();
  });

  test('HOMESCREEN_INTRO returns correct value', () => {
    expect(PresentationConstants.HOMESCREEN_INTRO).toMatchSnapshot();
  });

  test('HOMESCREEN_COUNTER_VALUE returns correct value', () => {
    expect(PresentationConstants.HOMESCREEN_COUNTER_VALUE).toMatchSnapshot();
  });

  test('HOMESCREEN_BUTTON_LABEL_BACK_TO_DEFAULT_FEATURE returns correct value', () => {
    expect(PresentationConstants.HOMESCREEN_BUTTON_LABEL_BACK_TO_DEFAULT_FEATURE).toMatchSnapshot();
  });

  test('HOMESCREEN_BUTTON_LABEL_ABOUT returns correct value', () => {
    expect(PresentationConstants.HOMESCREEN_BUTTON_LABEL_ABOUT).toMatchSnapshot();
  });

  test('ABOUT_WELCOME returns correct value', () => {
    expect(PresentationConstants.ABOUT_WELCOME).toMatchSnapshot();
  });

  test('ABOUT_BUTTON_LABEL_TOGGLE returns correct value', () => {
    expect(PresentationConstants.ABOUT_BUTTON_LABEL_TOGGLE).toMatchSnapshot();
  });

  test('ABOUT_HELLO_WORLD returns correct value', () => {
    expect(PresentationConstants.ABOUT_HELLO_WORLD).toMatchSnapshot();
  });

  test('ABOUT_GOODBYE_WORLD returns correct value', () => {
    expect(PresentationConstants.ABOUT_GOODBYE_WORLD).toMatchSnapshot();
  });
});
