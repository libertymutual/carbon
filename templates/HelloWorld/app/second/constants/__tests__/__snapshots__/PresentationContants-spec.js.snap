// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresentationConstants tests ABOUT_BUTTON_LABEL_TOGGLE returns correct value 1`] = `"Toggle Hello World Message"`;

exports[`PresentationConstants tests ABOUT_GOODBYE_WORLD returns correct value 1`] = `"Goodbye World"`;

exports[`PresentationConstants tests ABOUT_HELLO_WORLD returns correct value 1`] = `"Hello World"`;

exports[`PresentationConstants tests ABOUT_WELCOME returns correct value 1`] = `"Welcome To The Second Feature About Screen"`;

exports[`PresentationConstants tests CLOSE returns correct value 1`] = `"Close"`;

exports[`PresentationConstants tests CONTINUE returns correct value 1`] = `"Continue"`;

exports[`PresentationConstants tests GO_BACK returns correct value 1`] = `"Go Back"`;

exports[`PresentationConstants tests HOMESCREEN_BUTTON_LABEL_ABOUT returns correct value 1`] = `"About"`;

exports[`PresentationConstants tests HOMESCREEN_BUTTON_LABEL_BACK_TO_DEFAULT_FEATURE returns correct value 1`] = `"Back To Default Feature"`;

exports[`PresentationConstants tests HOMESCREEN_COUNTER_VALUE returns correct value 1`] = `"Counter Value:"`;

exports[`PresentationConstants tests HOMESCREEN_INTRO returns correct value 1`] = `
"This is an example of navigating to a completely separate React Native Feature

It also shows an example of sharing state between two separate React Native Features. The Counter Value below is the same counter value displayed/incremented/decremented on the Default Feature"
`;

exports[`PresentationConstants tests HOMESCREEN_WELCOME returns correct value 1`] = `"Welcome To The Second Feature Home Screen"`;

exports[`PresentationConstants tests NO returns correct value 1`] = `"No"`;

exports[`PresentationConstants tests OK returns correct value 1`] = `"Ok"`;

exports[`PresentationConstants tests PAGE_TITLE_ABOUT_SCREEN returns correct value 1`] = `"About"`;

exports[`PresentationConstants tests PAGE_TITLE_HOME_SCREEN returns correct value 1`] = `"Second Feature Home"`;

exports[`PresentationConstants tests YES returns correct value 1`] = `"Yes"`;
